go/sqltypes/value.go

```diff
@@ -547,6 +547,60 @@ func (v Value) EncodeSQLStringBuilder(b *strings.Builder) {
 	}
 }
 
+func (v Value) EncodeSQLStringBuilderWithCasting(b *strings.Builder) error {
+	castFn := func(typ string, quotes bool) {
+		b.WriteString("cast(")
```

Comment:
do we have test case for this method?

Reply:
it's used by all the end-to-end tests. are you looking for unit tests?